### PR TITLE
Share filter functionality to remove deleted or empty rp from seq, re…

### PR DESCRIPTION
**Pull request by darrylmurphy**
…move hasAccepted toggle from BankDetails

### CODE REVIEW

**Comment by darrylmurphy** [comment]
AMLS-3054

Bug created by lack of filter of deleted or empty. Solution will be to use this shared filter.

File: app/models/responsiblepeople/ResponsiblePeople.scala [hunk 1]
```diff
@@ -190,13 +190,13 @@ object ResponsiblePeople {
 
   implicit val formatOption = Reads.optionWithNull[Seq[ResponsiblePeople]]
 
+  def filter(rp: Seq[ResponsiblePeople]) = rp.filterNot(_.status.contains(StatusConstants.Deleted)).filterNot(_ == ResponsiblePeople())
```
Comment (elkdanger) [inline app/models/responsiblepeople/ResponsiblePeople.scala]: Now that the scope of this method has changed and is more generally available, could we rename it? Maybe something like `filterDeleted`, or `cleanup`?

I'm just conscious that `filter` doesn't really tell enough about what it filters. Do you have any strong opinions about it?

Comment (darrylmurphy) [reply app/models/responsiblepeople/ResponsiblePeople.scala]: Perhaps? Although it'll filter deleted and empty RPs at the moment, does filterDeleted stray from the truth and cleanup == filter?
Does it sound fair enough that "filter" used in a general context imply it'll remove RPs such as deleted and other unwanted criteria?

Comment (elkdanger) [reply app/models/responsiblepeople/ResponsiblePeople.scala]: I suppose as long as we're generally aware of what it does. Maybe if confusion arises from it in the future, we can look at refactoring it a bit to make it more clear.

Comment (darrylmurphy) [reply app/models/responsiblepeople/ResponsiblePeople.scala]: I agree, I think its trying to balance in the name between being descriptive but flexible. I certainly don't feel strongly about it, so I'm open to change if it proves unsuitable.

**Pull request merged**